airflow/jobs/scheduler_job.py

```diff
@@ -766,7 +766,17 @@ def _clean_tis_without_dagrun(self, session):
                 self._change_state_for_tis_without_dagrun(
                     old_states=[State.UP_FOR_RETRY], new_state=State.FAILED, session=session
                 )
-
+                
+                guard.commit()
```

Comment:
This should be outside of this try block.

Reply:
main idea here:
1) protect "commit" operation
2) split up code with two seprate actions, update status :
   2.1) mark as FAILED
   2.2) mark as None